Word completion suggestion interferes with website's suggestion results

rr-codes opened the pull request:
#### ab8b39b36339bb473f6c3f5943732a78d1d87d34
<pre>
Word completion suggestion interferes with website&apos;s suggestion results
<a href="https://bugs.webkit.org/show_bug.cgi?id=271197">https://bugs.webkit.org/show_bug.cgi?id=271197</a>
<a href="https://rdar.apple.com/124727588">rdar://124727588</a>

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

If the value of `writingsuggestions` isn&apos;t explicitly specified, have the default value for it be
`false` instead of `true` if the element has disabled `autocomplete`.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::isWritingSuggestionsEnabled const):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingSuggestions.mm:
(TEST):

Canonical link: <a href="https://commits.webkit.org/276344@main">https://commits.webkit.org/276344@main</a>
</pre>

aprotyas commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -4798,6 +4798,19 @@ bool Element::isWritingSuggestionsEnabled() const
+    // This is not yet part of the spec, but it improves web-compatibility; if autocomplete
+    // is intentionally off, the site author probably wants writingsuggestions off too.
+
+    for (auto* ancestor = this; ancestor; ancestor = ancestor->parentElementInComposedTree()) {
```
Per our smart ptr guidelines
```suggestion
    for (RefPtr ancestor = this; ancestor; ancestor = ancestor->parentElementInComposedTree()) {
```

pxlcoder commented on Source/WebCore/dom/Element.cpp:
```diff
@@ -4798,6 +4798,19 @@ bool Element::isWritingSuggestionsEnabled() const
+    // This is not yet part of the spec, but it improves web-compatibility; if autocomplete
+    // is intentionally off, the site author probably wants writingsuggestions off too.
+
+    for (RefPtr ancestor = this; ancestor; ancestor = ancestor->parentElementInComposedTree()) {
```
I don't think `autocomplete` is normally ancestor dependent.

rr-codes committed:
Word completion suggestion interferes with website's suggestion results
https://bugs.webkit.org/show_bug.cgi?id=271197
rdar://124727588

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

If the value of `writingsuggestions` isn't explicitly specified, have the default value for it be
`false` instead of `true` if the element has disabled `autocomplete`.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::isWritingSuggestionsEnabled const):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingSuggestions.mm:
(TEST):

Canonical link: https://commits.webkit.org/276344@main
